analytic_engine/src/sst/parquet/builder.rs

```diff
@@ -67,80 +64,72 @@ impl RecordBytesReader {
-            // Note: pending_record_batch may contains multiple batches
-            while fetched_row_num >= self.num_rows_per_row_group {
-                match pending_record_batch.pop_front() {
-                    // accumulated records is enough for one batch
-                    Some(next) if next.num_rows() >= remaining => {
-                        debug!(
-                            "enough target:{}, remaining:{}, fetched:{}, current:{}",
-                            self.num_rows_per_row_group,
-                            remaining,
-                            fetched_row_num,
-                            next.num_rows(),
-                        );
-                        current_batch.push(next.slice(0, remaining));
-                        let left = next.num_rows() - remaining;
-                        if left > 0 {
-                            pending_record_batch.push_front(next.slice(remaining, left));
-                        }
+        let mut prev_record_batch: Option<RecordBatchWithKey> = None;
 
-                        self.partitioned_record_batch
-                            .push(std::mem::take(&mut current_batch));
-                        fetched_row_num -= self.num_rows_per_row_group;
-                        remaining = self.num_rows_per_row_group;
-                    }
-                    // not enough for one batch
-                    Some(next) => {
-                        debug!(
-                            "not enough target:{}, remaining:{}, fetched:{}, current:{}",
-                            self.num_rows_per_row_group,
-                            remaining,
-                            fetched_row_num,
-                            next.num_rows(),
-                        );
-                        remaining -= next.num_rows();
-                        current_batch.push(next);
-                    }
-                    // nothing left, put back to pending_record_batch
-                    _ => {
-                        for records in std::mem::take(&mut current_batch) {
-                            pending_record_batch.push_front(records);
-                        }
-                        break;
-                    }
-                }
+        loop {
+            let row_group = self.fetch_next_row_group(&mut prev_record_batch).await?;
+            if row_group.is_empty() {
+                break;
             }
+            self.partitioned_record_batch.push(row_group);
         }
 
-        // collect remaining records into one batch
-        let mut remaining = Vec::with_capacity(pending_record_batch.len());
-        while let Some(batch) = pending_record_batch.pop_front() {
-            remaining.push(batch);
-        }
-        if !remaining.is_empty() {
-            self.partitioned_record_batch.push(remaining);
+        Ok(())
+    }
+
+    /// Fetch an integrate row group from the `self.record_stream`.
```

Comment:
What does `integrate row group` means?

Reply:
The word `integrate` may be a wrong usage here. Maybe `integral` or `complete` is better.

Actually, it means the number of the row group is `num_rows_per_group`.